Scripts/Runtime/Logging/.CSProject/UnityLogListener.cs

```diff
@@ -103,6 +106,20 @@ private void OnDestroy()
             }
         }
 
+        // When determining the context of a log, any frames in the stack trace matching these types are skipped
+        private static readonly string[] SKIPPED_STACK_FRAME_TYPES =
+        {
+            "System.Diagnostics.Debug",
+            "System.Diagnostics.Trace",
+            "System.Diagnostics.TraceInternal",
+            "UnityEngine.Debug",
+            "UnityEngine.Logger",
+            "UnityEngine.Assertions",
+            // Check for Unity's new stubbed out logging class that currently just proxies the existing logging system.
+            // This was found in the v0.17 of the Unity.Entities package
+            "Unity.Debug",
+        };
```

Comment:
As I ran into multiple flows used by Unity to send logs through its log handlers, I opted for a more flexible, if slightly less performant approach to resolving log contexts - rather that assume the flow will always be a fixed sequence of functions (especially in future Unity versions), I explicitly skip the types/namespaces listed here, as well as anything with Anvil's `Exclude` attribute.

The difference is a couple of `new StackFrame(...)` calls per log, which should not make a noticeable difference (that's something like 0.01ms), but if it's still a performance concern I can go back to making assumptions about Unity's logging call stack to improve it.

Reply:
Do you mean so I can say `SKIPPED_STACK_FRAME_TYPES.Contains(typeName)`? `Any` will short circuit when it finds a match, but I guess it always checks the whole array once when it finds the original context...

I think that works - I originally had `.StartsWith(typeName)` but I think I ended up with only full type names in this list (ex. initially I ignored anything under `System.Diagnostics.*`) so I could change it to a HashSet.Contains()

Reply:
I lied, `.StartsWith(...)` was important because "UnityEngine.Assertions" is a namespace, so I had to specify "UnityEngine.Assertions.Assert" (and "System.Diagnostics.TraceListener") - if we catch any other internal logging types showing up as the log context in the future, they'll just have to be added to this list.

Reply:
Ah ok, I had originally just read that it was checking if it was one rather than the start of one. I feel like this list will be updated pretty infrequently so it should be fine to just keep as the hashset and add to it as needed.

Reply:
Technically there's a risk that someone else defines types with the same name in the same namespace but a different assembly and we should check the source assembly of the type. That's just stupid behaviour though and I don't think we need to guard against it. This solution looks good.

Reply:
I'm in the "that's just stupid behaviour" boat, but even if you did add to `UnityEngine.Assertions` or something, that sounds like more code that should be skipped in determining log context anyway.